code/core/src/types/modules/addons.ts

```diff
@@ -474,6 +477,13 @@ export interface Addon_TestProviderType<
   name: string;
   title: (state: Addon_TestProviderState<Details>) => ReactNode;
   description: (state: Addon_TestProviderState<Details>) => ReactNode;
+  contextMenu?: (
+    options: {
+      context: API_HashEntry;
+      state: Addon_TestProviderState<Details>;
+    },
+    components: { ListItem: typeof ListItem }
+  ) => ReactNode;
```

Comment:
This is the added API to `Addon_TestProviderType`.

I'm personally a bit on the fence about the `components` bit, as the user is likely to need more components anyway. Adding more is an option, but @JReinhold has stronger opinions on this than I do.